src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -563,18 +578,78 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
             }
         }
 
+        /// <remarks>
+        /// This method is analogous to <see cref="SourceOrdinaryMethodSymbol.PartialMethodChecks(SourceOrdinaryMethodSymbol, SourceOrdinaryMethodSymbol, BindingDiagnosticBag)" />.
+        /// Whenever new checks are added to this method, the other method should also have those checks added, if applicable.
+        /// </remarks>
         private void PartialPropertyChecks(SourcePropertySymbol implementation, BindingDiagnosticBag diagnostics)
         {
             Debug.Assert(this.IsPartialDefinition);
             Debug.Assert((object)this != implementation);
             Debug.Assert((object?)this.OtherPartOfPartial == implementation);
 
-            // PROTOTYPE(partial-properties): check and diagnose all disallowed differences between parts.
-            // PROTOTYPE(partial-properties): we should likely report a warning when only one of the parts has an explicit access modifier for consistency with methods.
-            if (DeclaredAccessibility != implementation.DeclaredAccessibility)
+            if (!TypeWithAnnotations.Equals(implementation.TypeWithAnnotations, TypeCompareKind.AllIgnoreOptions))
+            {
+                diagnostics.Add(ErrorCode.ERR_PartialPropertyTypeDifference, implementation.GetFirstLocation());
+            }
+            else if (MemberSignatureComparer.ConsideringTupleNamesCreatesDifference(this, implementation))
+            {
+                diagnostics.Add(ErrorCode.ERR_PartialMemberInconsistentTupleNames, implementation.GetFirstLocation(), this, implementation);
+            }
+            else if (RefKind != implementation.RefKind)
```

Comment:
>else if [](http://example.com/codeflow?start=12&length=7)

Feels like type and ref kind differences should be reported independently (as we do for partial methods)